doc/doxygen/src/mainpage.md

```diff
@@ -111,7 +111,7 @@ core
 ----
 This directory contains the actual kernel. The kernel consists of the
 scheduler, inter-process-communication (messaging), threading, thread
-synchronization, and supporting data-structures and type definitions.
+synchronization, supporting data-structures and type definitions.
```

Comment:
grammarly says, we need a comma before "and"

Reply:
@miri64 @kaspar030, are ok with this proposed documentation update ?

Reply:
https://www.grammarly.com/blog/comma-before-and/

It is really a style choice per grammarly, so if that is the style you use I will add the comma.

Reply:
I personally use and prefer the Oxford comma (so an "," before "and") in enumerations. Mainly because German can be sometimes ambiguous with the placing of the "and" and the Oxford comma provides a good tool to lift this ambiguity.

Reply:
The thing about this sentence however is that I believe it was originally two recursive enumerations (I marked them with `[]` to make them clearer:

> The kernel consists of [the scheduler, inter-process-communication (messaging), threading, thread
synchronization, and [supporting data-structures and type definitions]].

So I'm not sure if plainly removing the "and" here is correct. How about

> > The kernel consists of the scheduler, inter-process-communication (messaging), threading, and thread
synchronization, as well as supporting data-structures and type definitions.